Enterprise-CMCS/cmcs-eregulations
# EREGCSC-1549 -- Homepage Left Nav ToC

Resolves [EREGCSC-1549](https://jiraent.cms.gov/browse/EREGCSC-1549)

**Description**
New Table of Contents element on homepage that is styles as a left nav menu.

**This pull request changes:**

- Adds header and footer to `homepage_prototype` template
- Creates LeftNavCollapse Vue component that is content agnostic and can accept any content via a `slot`
- Creates Table of Contents (ToC) Vue components to fetch Titles and ToC content and then render them as designed
- Creates new e2e Cypress test suite to test left nav behavior
- Consolidates Utility, API, and Filter methods from `eregs-vite` and `eregs-component-lib` projects into a single `utilities` folder that sits alongside the two projects
   - There are still some API methods left behind in the `eregs-component-lib` that need to be refactored to work with both projects.  A [story has been written](https://jiraent.cms.gov/browse/EREGCSC-1893) to do this.
- Updates paths in `prototype` so that it builds locally (even though it doesn't completely work)

**Steps to manually verify this change:**

1. Visit `homepage_test` on the [experimental deployment](https://bej353hjzc.execute-api.us-east-1.amazonaws.com/dev795/homepage_test/)
2. Ensure that left sidebar is populated with Title tabs and Tables of Contents that correspond to each Title
3. Click through to a Part
4. Poke around the rest of the site until you are satisfied it works as expected
